### ISSUE TITLE
Root location 

### ISSUE THREAD

**Issue by siddjellali** [bitnami/bitnami-docker-wordpress-nginx#43]
**Description**

After install my WP theme, i noticed some errors in the console tab related to root location.

![image](https://user-images.githubusercontent.com/3965823/116131795-1f62b080-a6cd-11eb-80db-a85c50d3c3ce.png)

**Steps to reproduce the issue:**

1. Install dockercompose
2. Install any wordpress theme
3. Go to customize page (for instance)

**Describe the results you received:**

Some pages are out of root location.

**Describe the results you expected:**

Stay in root location

**Additional information you deem important (e.g. issue happens only occasionally):**

It maybe related to relative path in php page but in any case root location HAVE to be the root folder otherwise it's a big security matter.

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           19.03.15
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        99e3ed8
 Built:             Sat Jan 30 03:11:43 2021
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.15
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       99e3ed8
  Built:            Sat Jan 30 03:18:13 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.9
  GitCommit:        ea765aba0d05254012b0b9e595e995c09186427f
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0

```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 65
  Running: 61
  Paused: 0
  Stopped: 4
 Images: 54
 Server Version: 19.03.15
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ea765aba0d05254012b0b9e595e995c09186427f
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.14.138-rancher
 Operating System: RancherOS v1.5.8
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 43.23GiB
 Name: vm-docker
 ID: ZY6C:4JZ4:RLQL:QAEP:E7SP:RWBS:ZXD7:QVQG:6QGC:OJZD:S5ZE:H3E6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

Thanks by advance for your support.

**Comment by siddjellali** [comment]
My https.conf, pretty simple.

```
# HTTPS Server
server {
    listen  8443 ssl;

    ssl_certificate            bitnami/certs/server.crt;
    ssl_certificate_key        bitnami/certs/server.key;
    ssl_session_cache          shared:SSL:1m;
    ssl_session_timeout        5m;
    ssl_ciphers                HIGH:!aNULL:!MD5;
    ssl_prefer_server_ciphers  on;

    root /opt/bitnami/wordpress;
    index index.php;

    location / {
       try_files $uri $uri/ /index.php?q=$uri&$args;
    }

    if (!-e $request_filename)
    {
      rewrite ^/(.+)$ /index.php?q=$1 last;
    }

    location ~ \.php$ {
      fastcgi_pass localhost:9000;
      fastcgi_index index.php;
      fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
      include fastcgi_params;
    }
}
```

**Comment by migruiz4** [comment]
Hi @siddjellali,

Thank you for reporting this issue.

There are a couple of things I'd like to clarify here.

Based on the error messages from your screenshot, the issue seems to be caused by a plugin called `ttg-core` trying to load `.js` files from `http://example.com/bitnami/wordpress/wp-content/...` instead of `http://example.com/wp-content`.

The folder `/opt/bitnami/wordpress/wp-content` is a symlink to the persistence directory `/opt/bitnami/wordpress/wp-content`
```bash
I have no name!@1d074ddee515:/opt/bitnami/wordpress$ ls -la
total 236
drwxrwxr-x  1 root root  4096 Apr 27 14:00 .
drwxrwxr-x  1 root root  4096 Apr 26 23:37 ..
-rw-rw-r--  1 root root     0 Apr 26 18:10 .buildcomplete
drwxrwxr-x  3 root root  4096 Apr 26 23:37 extra-varnish
-rw-rw-r--  1 root root   405 Apr 26 18:10 index.php
-rw-rw-r--  1 root root 19915 Apr 26 18:10 license.txt
drwxrwxr-x  2 root root  4096 Apr 26 23:37 licenses
-rw-rw-r--  1 root root  7345 Apr 26 18:10 readme.html
drwxr-xr-x  2 1001 root  4096 Apr 27 14:00 tmp
-rw-rw-r--  1 root root   711 Apr 26 18:10 wordpress-htaccess.conf
-rw-rw-r--  1 root root  7165 Apr 26 18:10 wp-activate.php
drwxrwxr-x  9 root root  4096 Apr 26 23:37 wp-admin
-rw-rw-r--  1 root root   351 Apr 26 18:10 wp-blog-header.php
-rw-rw-r--  1 root root  2328 Apr 26 18:10 wp-comments-post.php
-rw-rw-r--  1 root root  2913 Apr 26 18:10 wp-config-sample.php
lrwxrwxrwx  1 1001 root    32 Apr 27 14:00 wp-config.php -> /bitnami/wordpress/wp-config.php
lrwxrwxrwx  1 1001 root    29 Apr 27 14:00 wp-content -> /bitnami/wordpress/wp-content
-rw-rw-r--  1 root root  3939 Apr 26 18:10 wp-cron.php
drwxrwxr-x 25 root root 12288 Apr 26 23:37 wp-includes
-rw-rw-r--  1 root root  2496 Apr 26 18:10 wp-links-opml.php
-rw-rw-r--  1 root root  3313 Apr 26 18:10 wp-load.php
-rw-rw-r--  1 root root 44994 Apr 26 18:10 wp-login.php
-rw-rw-r--  1 root root  8509 Apr 26 18:10 wp-mail.php
-rw-rw-r--  1 root root 21125 Apr 26 18:10 wp-settings.php
-rw-rw-r--  1 root root 31328 Apr 26 18:10 wp-signup.php
-rw-rw-r--  1 root root  4747 Apr 26 18:10 wp-trackback.php
-rw-rw-r--  1 root root  3236 Apr 26 18:10 xmlrpc.php
```

What might be happening is that the plugin is using something similar to this to obtain the files route:
```
require_once wp_normalize_path( dirname( __FILE__ ) . '/lib/class-aricolor.php' );
```

This would return the real directory path and not the symlink path, causing the route to be `$host/bitnami/wordpress/wp-content/...` which doesn't exist under the root path. 

This issue can be fixed by adding a rewrite rule to your Nginx configuration as shown below:
```
# HTTPS Server
server {
    listen  8443 ssl;

    ssl_certificate            bitnami/certs/server.crt;
    ssl_certificate_key        bitnami/certs/server.key;
    ssl_session_cache          shared:SSL:1m;
    ssl_session_timeout        5m;
    ssl_ciphers                HIGH:!aNULL:!MD5;
    ssl_prefer_server_ciphers  on;

    root /opt/bitnami/wordpress;
    index index.php;
    rewrite ^/bitnami/wordpress/wp-content/(.*)$ /wp-content/$1 last;

    location / {
       try_files $uri $uri/ /index.php?q=$uri&$args;
    }

    if (!-e $request_filename)
    {
      rewrite ^/(.+)$ /index.php?q=$1 last;
    }

    location ~ \.php$ {
      fastcgi_pass localhost:9000;
      fastcgi_index index.php;
      fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
      include fastcgi_params;
    }
}
```

**Comment by siddjellali** [comment]
Your workarround works ! Thanks a lot your support !!!

`rewrite ^/bitnami/wordpress/wp-content/(.*)$ /wp-content/$1 last;`

Is it something dedicated for my issue or something you should add in the documentation 'as default value'.

**Comment by migruiz4** [comment]
I'm glad it worked!

We are evaluating how frequent this issue is and considering the benefits and drawbacks of adding a permanent fix to the Nginx configuration sample. 
We also noticed this may happen when using our WordPress + Apache image, so we will probably add an equivalent fix for it too.

Thank you again for reporting this issue and helping us improve our images.

**Comment by marcosbc** [comment]
Hi, just wanted to let  you know that we recently released [5.7.1-debian-10-r22](https://github.com/bitnami/bitnami-docker-wordpress-nginx#571-debian-10-r22) which includes an explicit fix for this issue, amongst lots of other improvements and fixes.
